tests/Tests/200__monitor_and_manage/200__metrics/200__metrics.robot

```diff
@@ -1,26 +1,136 @@
 
 Check Prometheus Alerting Rules
-  Prometheus.Verify Rules  ${RHODS_PROMETHEUS_URL}  ${RHODS_PROMETHEUS_TOKEN}  alert  @{ALERT_GROUPS}
+    Prometheus.Verify Rules    ${RHODS_PROMETHEUS_URL}    ${RHODS_PROMETHEUS_TOKEN}    alert    @{ALERT_GROUPS}
+
+Read Current CPU Usage
+    [Documentation]    Returns list of current cpu usage
+    ${version_check} =    Is RHODS Version Greater Or Equal Than    1.5.0
+    IF    ${version_check}==True
+        ${Expression} =    Set Variable
+        ...    sum(rate(container_cpu_usage_seconds_total{prometheus_replica="prometheus-k8s-0", container="",pod=~"jupyterhub-nb.*",namespace="rhods-notebooks"}[1h]))
+    ELSE
+        ${Expression} =    Set Variable
+        ...    sum(rate(container_cpu_usage_seconds_total{prometheus_replica="prometheus-k8s-0", container="",pod=~"jupyterhub-nb.*",namespace="rhods-notebooks"}[5m]))
+    END
+    ${resp} =    Prometheus.Run Query    ${RHODS_PROMETHEUS_URL}    ${RHODS_PROMETHEUS_TOKEN}    ${Expression}
+    [Return]    ${resp.json()["data"]["result"][0]["value"][-1]}
+
+## TODO: Add this keyword with the other JupyterHub stuff
+
+Run Jupyter Notebook For 5 Minutes
+    Open Browser    ${ODH_DASHBOARD_URL}    browser=${BROWSER.NAME}    options=${BROWSER.OPTIONS}
+    Login To RHODS Dashboard    ${TEST_USER.USERNAME}    ${TEST_USER.PASSWORD}    ${TEST_USER.AUTH_TYPE}
+    Wait for RHODS Dashboard to Load
+    Iterative Image Test    s2i-generic-data-science-notebook    https://github.com/lugi0/minimal-nb-image-test
+    ...    minimal-nb-image-test/minimal-nb.ipynb
+
+##TODO: This is a copy of "Iterative Image Test" keyword from image-iteration.robob. We have to refactor the code not to duplicate this method
+
+Iterative Image Test
+    [Arguments]    ${image}    ${REPO_URL}    ${NOTEBOOK_TO_RUN}
+    ${version-check} =    Is RHODS Version Greater Or Equal Than    1.4.0
+    IF    ${version-check}==True
+        Launch JupyterHub From RHODS Dashboard Link
+    ELSE
+        Launch JupyterHub From RHODS Dashboard Dropdown
+    END
+    Login To Jupyterhub    ${TEST_USER.USERNAME}    ${TEST_USER.PASSWORD}    ${TEST_USER.AUTH_TYPE}
+    Page Should Not Contain    403 : Forbidden
+    ${authorization_required} =    Is Service Account Authorization Required
+    Run Keyword If    ${authorization_required}    Authorize jupyterhub service account
+    Fix Spawner Status
+    Spawn Notebook With Arguments    image=${image}
+    Run Cell And Check Output    print("Hello World!")    Hello World!
+    Capture Page Screenshot
+    JupyterLab Code Cell Error Output Should Not Be Visible
+    #This ensures all workloads are run even if one (or more) fails
+    Run Keyword And Continue On Failure    Clone Git Repository And Run    ${REPO_URL}    ${NOTEBOOK_TO_RUN}
+    Clean Up Server
+    Stop JupyterLab Notebook Server
+    Go To    ${ODH_DASHBOARD_URL}
+    Sleep    10
+
+CleanUp JupyterHub
```

Comment:
where are you using this kw? 
are you sure isn't it already defined somewhere in the jupyter resources?

Reply:
Thanks I have not used , I will remove that one